### PR TITLE
Enhance LoDResetOp and add python wrapper

### LINKED ISSUE #9171: CopyLoDOp is needed.
Sometimes we need set a LoDTensor's LoD by copying LoD from another LoDTensor. If the referred LoDTensor has more than one LoD level, the level range should be specified by attribute. If CopyLoDOp provided, we can view a common Tensor as a sequence by assigning a LoD information to it. 

For example, `beam_search` operator requires that `prev_ids`, `topk_indices`, and `topk_scores` are all LoDTensor, however, LoD of `topk_scores` usually  inherits from `states`,  `states` sometimes is a common Tensor, since the init value is always Tensor. In this situation, we can copy LoD from `prev_ids` to `topk_scores`.

### PR DESCRIPTION
**Pull request by pkuyym**
Resolves #9171

### CODE REVIEW

File: paddle/fluid/operators/lod_reset_op.cc [hunk 1]
```diff
@@ -50,36 +49,42 @@ class LoDResetOpMaker : public framework::OpProtoAndCheckerMaker {
...
+target_lod: [0, 4, 6]
 
-If target LoD = [0, 4, 6], the lod_reset operator will reset the LoD and
-the sequences that the LoDTensor Output(Out) contains becomes:
-
-    [1.0, 2.0, 3.0, 4.0], [5.0, 6.0].
+then we get an 1-level LoDTensor
+    Out.lod =  [[ 0,                   4,            6 ]]
+    Out.data = [[1.0], [2.0], [3.0], [4.0], [5.0], [6.0]]
+    Out.dims = [6, 1]
```
Comment (qingqing01) [inline paddle/fluid/operators/lod_reset_op.cc]: Add example for Y.

File: paddle/fluid/operators/lod_reset_op.cc [hunk 2]
```diff
@@ -50,36 +49,42 @@ class LoDResetOpMaker : public framework::OpProtoAndCheckerMaker {
...
-    AddInput("X", "(LoDTensor) The input tensor of lod_reset operator.");
-    AddInput("TargetLoD",
-             "(Tensor, optional) The target level 0 LoD from Input().")
+    AddInput("X",
+             "(Tensor, LoDTensor) Input variable of LoDResetOp which "
+             "could be a Tensor or LoDTensor, where the data of output "
+             "variable inherits from.");
+    AddInput("Y",
+             "(Tensor, LoDTensor, optional) If provided, lod of Input(Y) would "
+             "be considered as the target lod first, otherwise data of "
```
Comment (qingqing01) [inline paddle/fluid/operators/lod_reset_op.cc]: If provided and Y is LoDTensor, ..., otherwise ?

Comment (pkuyym) [reply paddle/fluid/operators/lod_reset_op.cc]: Done.

File: paddle/fluid/operators/lod_reset_op.cc (hunk 1 above)
Comment (pkuyym) [reply paddle/fluid/operators/lod_reset_op.cc]: Done.

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -3292,3 +3293,98 @@ def autoincreased_step_counter(counter_name=None, begin=1, step=1):
         counter.stop_gradient = True
 
     return counter
+
+
+def lod_reset(x, y, target_lod=None):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]:
```python
def lod_reset(x, y=None, target_lod=None):
```

Comment (pkuyym) [reply python/paddle/fluid/layers/nn.py]: y == None and target_lod == None? Seems that we prefer target lod from y.

**Pull request merged**